src/internal/agent/hooks/common.go

```diff
@@ -13,3 +27,43 @@ func getLabelPatch(currLabels map[string]string) operations.PatchOperation {
 	currLabels["zarf-agent"] = "patched"
 	return operations.ReplacePatchOperation("/metadata/labels", currLabels)
 }
+
+func getManifestConfigMediaType(ctx context.Context, zarfState *state.State, imageAddress string) (string, error) {
+	client := &auth.Client{
+		Client: orasRetry.DefaultClient,
+		Cache:  auth.NewCache(),
+		Credential: auth.StaticCredential(imageAddress, auth.Credential{
+			Username: zarfState.RegistryInfo.PullUsername,
+			Password: zarfState.RegistryInfo.PullPassword,
+		}),
+	}
+
+	ref, err := registry.ParseReference(imageAddress)
+	if err != nil {
+		return "", err
+	}
+
+	http, err := images.ShouldUsePlainHTTP(ctx, ref.Registry, client)
+	if err != nil {
+		return "", err
+	}
```

Comment:
Easy to miss but the problem is that we're using the imageAddress rather than the registry address for credentials. ORAS requires that staticCredentials only take the host. 
```suggestion
	ref, err := registry.ParseReference(imageAddress)
	if err != nil {
		return "", err
	}
	client := &auth.Client{
		Client: orasRetry.DefaultClient,
		Cache:  auth.NewCache(),
		Credential: auth.StaticCredential(ref.Registry, auth.Credential{
			Username: zarfState.RegistryInfo.PullUsername,
			Password: zarfState.RegistryInfo.PullPassword,
		}),
	}

	http, err := images.ShouldUsePlainHTTP(ctx, ref.Registry, client)
	if err != nil {
		return "", err
	}
```